syft/file/classification_cataloger.go

```diff
@@ -0,0 +1,38 @@
+package file
+
+import (
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/source"
+)
+
+type ClassificationCataloger struct {
+	classifiers []Classifier
+}
+
+func NewClassificationCataloger(classifiers []Classifier) (*ClassificationCataloger, error) {
+	return &ClassificationCataloger{
+		classifiers: classifiers,
+	}, nil
+}
+
+func (i *ClassificationCataloger) Catalog(resolver source.FileResolver) (map[source.Location][]Classification, error) {
+	results := make(map[source.Location][]Classification)
+
+	numResults := 0
+	for location := range resolver.AllLocations() {
```

Comment:
Open question (for longer term) — do we think it's worth considering a slice-returning method on the resolver, if consumers are having to count out enumerated results?

Reply:
the thought was if there are several locations, a chan would generate them one at a time, so a caller doesn't have to hold all locations in memory --we have multiple callers here in parallel. This interface method has come up a few times in the same way in PR comments ("why a channel over a slice?")... so it sounds like there is an overwhelming desire to make this a slice

Reply:
Okay, cool. Maybe at least as another option. If other consumers are getting value specifically from the `chan` approach, that'd be good to leave. No change needed in this PR!